editor/store/reducer.js

```diff
@@ -1091,6 +1099,99 @@ export function autosave( state = null, action ) {
 	return state;
 }
 
+/**
+ * Reducer managing the block types
+ *
+ * @param {Object} state  Current state.
+ * @param {Object} action Dispatched action.
+ *
+ * @return {Object} Updated state.
+ */
+export function tokens( state = {}, action ) {
+	switch ( action.type ) {
+		case 'REGISTER_TOKEN': {
+			const { name } = action;
+
+			if ( typeof name !== 'string' ) {
+				error(
```

Comment:
Interesting that you chose to do the validation in the reducers. I think in redux's theory, the reducers should be pure (free of side effects) to be able to run them multiple times, rewind etc... without side effects. I wonder if we should move the validation into the `registerToken` action creator instead (by allowing `null` being returned from action creators). cc @aduth

Reply:
So should I move the validation to the action creators, or would it not belong there either? Should it me moved back to separate functions? Do the token settings even belong in the `core/editor` store at all?

Reply:
I think action creators is the best place for it but I'm not certain the data module is ready to handle "empty actions" but should be easy to update to avoid calling `dispatch` on `null` actions.

Reply:
Okay, let's keep a wrapper function for now.